### PR TITLE
Introduce the number formatting settings

### LINKED ISSUE #5108: [FEATURE] Add numeric formatter "fraction" style
Original commit: https://github.com/qgis/QGIS/commit/78d445ee654b54937344da61f6c933732405d836 by nyalldawson

This style represents decimal numbers as vulgar fractions, e.g.
"3/4" instead of 0.75.

Options include using Unicode superscript and subscript characters
for nicer typography, e.g. ¹⁷/₂₃ (this is the default mode, disabling
this option uses the "17/23" format). An option also exists for
using dedicated unicode characters for specific fractions (where
a unicode character exists), e.g. ½ or ¾

Ultimately this allows for creation of scalebars with fractional
representations of distances, e.g. 0 ----- ½ ----- 1 km
(instead of 0 ------ 0.5 ------ 1km)

Fixes #21289

Sponsored by SLYR

### PR DESCRIPTION
**Pull request by DelazJ**
~~I did not see a report for this in the repo but~~ it refs https://github.com/qgis/QGIS/pull/33624
Found! Fixes #5108

### CODE REVIEW

File: docs/user_manual/introduction/general_tools.rst [hunk 1]
```diff
@@ -2220,6 +2220,60 @@ screen resolution, paper size, or the terrain). Available units are:
      Adjust scaling range dialog
 
 
+.. index:: Number format; Map scale bar
+.. _number_formatting:
+
+Number Formatting
+-----------------
+
+A numeric formatter allows for formatting a numeric value for display, using
```
Comment (havatv) [inline docs/user_manual/introduction/general_tools.rst:2229]:
```suggestion
Numeric formatters allow formatting of numeric values for display, using
```

File: docs/user_manual/introduction/general_tools.rst [hunk 2]
```diff
@@ -2220,6 +2220,60 @@ screen resolution, paper size, or the terrain). Available units are:
...
+a variety of different formatting techniques (e.g. as scientific notation,
```
Comment (havatv) [inline docs/user_manual/introduction/general_tools.rst:2230]:
```suggestion
a variety of different formatting techniques (for instance scientific notation,
```

File: docs/user_manual/introduction/general_tools.rst [hunk 3]
```diff
@@ -2220,6 +2220,60 @@ screen resolution, paper size, or the terrain). Available units are:
...
+currency values, percentage values, etc). This can be used for example when
```
Comment (havatv) [inline docs/user_manual/introduction/general_tools.rst:2231]:
```suggestion
currency values, percentage values, etc). One use of this is to set
```

File: docs/user_manual/introduction/general_tools.rst [hunk 4]
```diff
@@ -2220,6 +2220,60 @@ screen resolution, paper size, or the terrain). Available units are:
...
+setting text in a layout scale bar or fixed table...
```
Comment (havatv) [inline docs/user_manual/introduction/general_tools.rst:2232]:
```suggestion
text in a layout scale bar or fixed table.
```

File: docs/user_manual/introduction/general_tools.rst [hunk 5]
```diff
@@ -2220,6 +2220,60 @@ screen resolution, paper size, or the terrain). Available units are:
...
+Different categories of formats are supported.
+For most of them, you can set part or all of the following numeric options:
+
+* |checkbox| :guilabel:`Show thousands separator`
+* |unchecked| :guilabel:`Show plus sign`
+* |unchecked| :guilabel:`Show trailing zeros`
+
+But they can also have their custom settings. Provided categories are:
+
+* :guilabel:`General`, the default category, has no setting and displays values
```
Comment (havatv) [inline docs/user_manual/introduction/general_tools.rst:2243]:
```suggestion
* :guilabel:`General`, the default category. Has no setting, and displays values
```

File: docs/user_manual/introduction/general_tools.rst [hunk 6]
```diff
@@ -2220,6 +2220,60 @@ screen resolution, paper size, or the terrain). Available units are:
...
+  as set in the parent widget properties or using the global settings
```
Comment (havatv) [inline docs/user_manual/introduction/general_tools.rst:2244]:
```suggestion
  as set in the parent widget properties or using the global settings.
```

File: docs/user_manual/introduction/general_tools.rst [hunk 7]
```diff
@@ -2220,6 +2220,60 @@ screen resolution, paper size, or the terrain). Available units are:
...
+
+  * :guilabel:`Format`: possible ranges of values are ``0 to 180°, with E/W suffix``,
+    ``-180 to +180°`` and ``0 to 360°``
+  * number of :guilabel:`Decimal places`
+* :guilabel:`Currency` for a text representation of a currency value.
+
+  * :guilabel:`Prefix`
+  * :guilabel:`Suffix`
+  * number of :guilabel:`Decimal places`
+* :guilabel:`Fraction` for a vulgar fractional representation of a decimal
```
Comment (havatv) [inline docs/user_manual/introduction/general_tools.rst:2260]: Why"vulgar"?

File: docs/user_manual/introduction/general_tools.rst [hunk 8]
```diff
@@ -2220,6 +2220,60 @@ screen resolution, paper size, or the terrain). Available units are:
...
+  value (e.g. *1/2* instead of *0.5*)
+
+  * |unchecked| :guilabel:`Use unicode super/subscript` to show eg :sup:`1/2`
```
Comment (havatv) [inline docs/user_manual/introduction/general_tools.rst:2263]:
```suggestion
  * |unchecked| :guilabel:`Use unicode super/subscript` to show. For example :sup:`1/2`
```

File: docs/user_manual/introduction/general_tools.rst [hunk 9]
```diff
@@ -2220,6 +2220,60 @@ screen resolution, paper size, or the terrain). Available units are:
...
+    instead of 1/2
+  * |unchecked| :guilabel:`Use dedicated Unicode characters`
+  * customize the :guilabel:`Thousands separator`
+* :guilabel:`Percentage` appending ``%`` to the values
+
+  * number of :guilabel:`Decimal places`
+  * :guilabel:`Scaling` to indicate whether the actual values already represent
+    percentages (and they will be kept as is) or fractions (and they are converted)
```
Comment (havatv) [inline docs/user_manual/introduction/general_tools.rst:2271]:
```suggestion
    percentages (then they will be kept as is) or fractions (then they are converted)    
```

File: docs/user_manual/introduction/general_tools.rst [hunk 10]
```diff
@@ -2220,6 +2220,60 @@ screen resolution, paper size, or the terrain). Available units are:
...
+  * |unchecked| :guilabel:`Use unicode super/subscript` to show eg :sup:`1/2`
+    instead of 1/2
+  * |unchecked| :guilabel:`Use dedicated Unicode characters`
+  * customize the :guilabel:`Thousands separator`
+* :guilabel:`Percentage` appending ``%`` to the values
+
+  * number of :guilabel:`Decimal places`
+  * :guilabel:`Scaling` to indicate whether the actual values already represent
+    percentages (and they will be kept as is) or fractions (and they are converted)    
+* :guilabel:`Scientific` notation in the form ``2.56e+03``. A number of
```
Comment (havatv) [inline docs/user_manual/introduction/general_tools.rst:2272]:
```suggestion
* :guilabel:`Scientific` notation in the form ``2.56e+03``. The number of
```

File: docs/user_manual/introduction/general_tools.rst (hunk 5 above)
Comment (DelazJ) [reply docs/user_manual/introduction/general_tools.rst:2243]: Looks weird to me to have a sentence beginning with a verb, and no subject. Oh! I did it. was weird, right?

File: docs/user_manual/introduction/general_tools.rst (hunk 7 above)
Comment (DelazJ) [reply docs/user_manual/introduction/general_tools.rst:2260]: Probably because of the native English speaker PR author... 😄

Comment (havatv) [reply docs/user_manual/introduction/general_tools.rst:2260]: I looked it up - seems to be a standard term.

File: docs/user_manual/introduction/general_tools.rst (hunk 5 above)
Comment (havatv) [reply docs/user_manual/introduction/general_tools.rst:2243]: A dash or a colon are possible alternatives.
`:guilabel:`General`, the default category - has no setting, and displays values`
`:guilabel:`General`, the default category: has no setting, and displays values`

**Pull request merged**